pkg/function/service.go

```diff
@@ -208,23 +183,24 @@ func (fs *Service) ReduceFn(stream functionpb.UserDefinedFunction_ReduceFnServer
 			// TODO: research on gRPC errors and revisit the error handler
 			return err
 		}
+		unifiedKey := strings.Join(d.GetKeys(), "")
```

Comment:
This means `["aa", "bb"]` will be grouped together with `["aab", "b"]`?

Reply:
the lookup key will be "aabb"

Reply:
Are we supposed to use different channels for `["aa", "bb"]` and `["aab", "b"]`?

Reply:
that is a very fair point. we should 🙂